src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_DashboardStack/Dashboard_TripSearchScreen/use-get-service-journey-vehicles.ts

```diff
@@ -0,0 +1,42 @@
+import {getServiceJourneyVehicles} from '@atb/api/vehicles';
+import {useCallback, useEffect, useState} from 'react';
+import usePollableResource from '@atb/utils/use-pollable-resource';
+import {AxiosError} from 'axios';
+import {GetServiceJourneyVehicles} from '@atb/api/types/generated/ServiceJourneyVehiclesQuery';
+
+export const useGetServiceJourneyVehicles = (serviceJourneyIds?: string[]) => {
+  const [vehicleUpdates, setVehicleUpdates] = useState<{
+    serviceJourneyVehicles?: GetServiceJourneyVehicles;
+    error?: AxiosError;
+  }>();
+
+  const fetchVehicles = useCallback(
+    (signal?: AbortSignal) =>
+      getServiceJourneyVehicles(serviceJourneyIds, {
+        signal,
+      }),
+    [JSON.stringify(serviceJourneyIds)],
+  );
+
+  const [updatedVehicles, , , error] = usePollableResource<
+    GetServiceJourneyVehicles | undefined,
+    AxiosError
+  >(fetchVehicles, {
+    initialValue: undefined,
+    pollingTimeInSeconds: 20,
+  });
+
+  useEffect(
+    () =>
+      setVehicleUpdates({
+        serviceJourneyVehicles: updatedVehicles,
+        error: error,
+      }),
+    [updatedVehicles, error],
+  );
```

Comment:
Is this `useEffect` and the `useState` really needed? Can't we just use the state directly from `usePollableResource` without having this intermediate storage?